app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,83 @@
+package org.mozilla.fenix.components
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.IntentSender
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.mozilla.fenix.home.intent.StartSearchIntentProcessor
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PrivateShortcutCreateManagerTest {
+
+    @Test
+    fun `GIVEN shortcut pinning is not supported WHEN createPrivateShortcut is called THEN do not create a pinned shortcut`() {
+        val privateShortcutCreateManager = spyk<PrivateShortcutCreateManager>()
+        mockkStatic(ShortcutManagerCompat::class)
+
+        every { ShortcutManagerCompat.isRequestPinShortcutSupported(testContext) } returns false
+
+        privateShortcutCreateManager.createPrivateShortcut(testContext)
+
+        verify(exactly = 0) { ShortcutManagerCompat.requestPinShortcut(testContext, any(), any()) }
+    }
+
+    @Test
+    fun `GIVEN shortcut pinning is supported WHEN createPrivateShortcut is called THEN create a pinned shortcut`() {
+        val shortcut = slot<ShortcutInfoCompat>()
+        val intentSender = slot<IntentSender>()
+        val intent = slot<Intent>()
+
+        val privateShortcutCreateManager = spyk<PrivateShortcutCreateManager>()
+        mockkStatic(ShortcutManagerCompat::class)
```

Comment:
This static mocks can be put in a try-catch-finally block.

Reply:
went with `@Before and @After approach`